src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+    }
+
+    if (animal.health <= 0) {
+      const index = Animal.alive.indexOf(animal);
+
+      if (index > -1) {
+        Animal.alive.splice(index, 1);
+      }
```

Comment:
[CODE PERFORMANCE] - Instead of using `indexOf` and `splice` within a loop, which can be inefficient, consider using the `filter` method to remove the dead animal. This can be done by reassigning `Animal.alive` to the filtered array where only animals with health greater than 0 are kept.